src/java/com/sarality/app/view/action/ShowAlertDialogAction.java

```diff
@@ -0,0 +1,46 @@
+package com.sarality.app.view.action;
+
+import android.app.AlertDialog;
+import android.content.Context;
+import android.view.LayoutInflater;
+import android.view.View;
+
+import com.sarality.app.view.list.ViewRenderer;
+
+/**
+ * Action that would show the Alert Dialog with a message
+ *
+ * @author sunayna@ (Sunayna Uberoy)
+ */
+public class ShowAlertDialogAction<T> extends BaseViewAction implements ViewAction {
+  private final Context context;
+  private final int layout;
+  private final ViewRenderer<T> renderer;
+
+  /**
+   * Constructor.
+   *
+   * @param viewId Id of the View that triggers the action.
+   * @param triggerType Type of event that triggers the action.
+   */
+  public ShowAlertDialogAction(int viewId, TriggerType triggerType, Context context, int layout,
+                               ViewRenderer<T> renderer) {
```

Comment:
This is fine for now but I have a feeling that we'll need an initializer here rather than a Renderer.

A ViewInitializer has the advantage that if can load the data that it may need and then call the render method.